modules/notify/src/main/NotificationRepo.scala

```diff
@@ -29,8 +29,9 @@ final private class NotificationRepo(colls: NotifyColls)(using Executor):
   def markManyRead(doc: Bdoc): Funit =
     coll.update.one(doc, $set("read" -> true), multi = true).void
 
-  def unreadNotificationsCount(userId: UserId): Fu[UnreadCount] =
-    UnreadCount from coll.countSel(unreadOnlyQuery(userId))
+  def expireAndCount(userId: UserId): Fu[UnreadCount] =
+    markManyRead(expiredQuery(userId)) flatMap: _ =>
+      UnreadCount from coll.countSel(unreadOnlyQuery(userId))
```

Comment:
I think automatic expiration from mongodb would be more efficient https://www.mongodb.com/docs/manual/tutorial/expire-data/

Reply:
Could we deploy "mark as read" as it exists in this PR (including any performance improvements you can think of), and keep a close eye on it? If it's a measurable impact on load we can also back it out. As a user, I don't get many notifications so I'm not equipped to make the call. But it's possible yanking them from the inbox entirely may be less popular than having a misleading number on the badge, as even stale notifications provide some info to the user (in a historical, here's what happened while you were away, sense).